frontend/components/shelter/pet/add/FormComponents.tsx

```diff
@@ -313,23 +313,33 @@ const ImageGallery = ({
 	const imageUploadRef = useRef<HTMLInputElement>(null);
 	const [pickedImage, setPickedImage] = useState<File | null>(null);
 	const [cloned, setCloned] = useState([...images]);
-
+	
 	useEffect(() => {
 		if (!pickedImage) return;
-		const reader = new FileReader();
-		reader.onloadend = () => {
+		const waitForLoadedImage = (_pickedImage: File): Promise<string> => {
```

Comment:
Converted to async function. Would be nice to handle all the events reader would emit, but works for the happy case.